lib/src/arithmetic/floating_point/floating_point_adder_simple.dart

```diff
@@ -16,17 +16,16 @@ class FloatingPointAdderSimple extends FloatingPointAdder {
   /// Add two floating point numbers [a] and [b], returning result in [sum].
   /// - [adderGen] is an adder generator to be used in the primary adder
   /// functions.
-  /// - [ppTree] is an parallel prefix tree generator to be used in internal
-  /// functions.
+  /// - [priorityGen] is a [PriorityEncoder] generator to be used in the
+  /// leading one detection (default [RecursiveModulePriorityEncoder]).
   FloatingPointAdderSimple(super.a, super.b,
       {super.clk,
       super.reset,
       super.enable,
       Adder Function(Logic a, Logic b, {Logic? carryIn}) adderGen =
           NativeAdder.new,
-      ParallelPrefix Function(
-              List<Logic> inps, Logic Function(Logic term1, Logic term2) op)
-          ppTree = KoggeStone.new,
+      PriorityEncoder Function(Logic bitVector, {Logic? valid, String name})
```

Comment:
why require the function to support `name` if you're not using it?  probably good to keep it, but then use the name argument.